From 45d6dc184862adba678743097f8889d8dc08ba7a Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <ondrej.jirman@zonio.net>
Date: Thu, 20 Mar 2008 20:48:14 +0100
Subject: [PATCH] add CAL_STATIC_CAPABILITY_NO_SEND_IMIP calendar capability

CAL_STATIC_CAPABILITY_NO_SEND_IMIP

  Don't send iMIPs via Camel. Assume that backend will send it.
  Don't convert ATTACH properties to the CID:... before passing iTIP to
  itip_send_comp() in real_send_comp().
---
 calendar/gui/dialogs/comp-editor.c |    2 +-
 calendar/gui/itip-utils.c          |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/calendar/gui/dialogs/comp-editor.c b/calendar/gui/dialogs/comp-editor.c
index 17b2409..28278df 100644
--- a/calendar/gui/dialogs/comp-editor.c
+++ b/calendar/gui/dialogs/comp-editor.c
@@ -2613,7 +2613,7 @@ real_send_comp (CompEditor *editor, ECalComponentItipMethod method)
 			set_attendees_for_delegation (send_comp, address, method);
 	}
 
-	if (!e_cal_component_has_attachments (priv->comp)) {
+	if (!e_cal_component_has_attachments (priv->comp) || e_cal_get_static_capability (priv->client, CAL_STATIC_CAPABILITY_NO_SEND_IMIP)) {
 		if (itip_send_comp (method, send_comp, priv->client,
 					NULL, NULL, users)) {
 			g_object_unref (send_comp);
diff --git a/calendar/gui/itip-utils.c b/calendar/gui/itip-utils.c
index f12874b..f8be07b 100644
--- a/calendar/gui/itip-utils.c
+++ b/calendar/gui/itip-utils.c
@@ -1264,7 +1264,7 @@ itip_send_comp (ECalComponentItipMethod method, ECalComponent *send_comp,
 	/* Recipients */
 	to_list = comp_to_list (method, comp, users, FALSE);
 	if (method != E_CAL_COMPONENT_METHOD_PUBLISH) {
-		if (to_list == NULL || to_list->_length == 0) {
+		if (to_list == NULL || to_list->_length == 0 || e_cal_get_static_capability (client, CAL_STATIC_CAPABILITY_NO_SEND_IMIP)) {
 			/* We sent them all via the server */
 			retval = TRUE;
 			goto cleanup;
-- 
1.5.4-dirty

